casatables/src/glue.cc

```diff
@@ -131,8 +133,130 @@ extern "C" {
         return -1;
     }
 
+    // TableDesc
+
+    GlueTableDesc *
+    tabledesc_create(
+        // TODO: figure out exactly what this is.
+        // Apparently this is actually the name of the table?
+        const StringBridge &type
+    )
+    {
+        // TODO: expose this?
+        GlueTableDesc::TDOption tdOption = GlueTableDesc::TDOption::New;
+        return new GlueTableDesc(bridge_string(type), tdOption);
+    }
+
+    GlueTableDesc *
+    tabledesc_add_scalar_column(
+        GlueTableDesc &tableDesc,
+        GlueDataType dataType,
+        const StringBridge &columnName,
+        ExcInfo &exc
+    )
+    {
+        try {
+
+            switch (dataType) {
+
+#define CASE(DTYPE, CPPTYPE) \
+            case casacore::DTYPE: { \
+                tableDesc.addColumn(casacore::ScalarColumnDesc<CPPTYPE>( \
+                    bridge_string(columnName) \
+                )); \
+                break; \
+            }
+
+            CASE(TpBool, casacore::Bool)
+            CASE(TpChar, casacore::Char)
+            CASE(TpUChar, casacore::uChar)
+            CASE(TpShort, casacore::Short)
+            CASE(TpUShort, casacore::uShort)
+            CASE(TpInt, casacore::Int)
+            CASE(TpUInt, casacore::uInt)
+            CASE(TpFloat, float)
+            CASE(TpDouble, double)
+            CASE(TpComplex, casacore::Complex)
+            CASE(TpDComplex, casacore::DComplex)
+
+#undef CASE
+
+            case casacore::TpString:
+                throw std::runtime_error("use table_get_scalar_column_data_string for TpString columns");
```

Comment:
This error message needs updating. Or can we handle the TpString case like the other scalar column types here? Since there's no data value going back and forth, I feel like strings might work here.

Reply:
awesome, I've pushed and tested a fix for this, thanks!